### DISCUSSION TITLE
Generate PDF using Chromium/Firefox

### DISCUSSION THREAD

**Discussion by pguerra-ce** [hardkoded/puppeteer-sharp#2711] [Q&A]
I'm trying to generate a PDF in Linux Arm64. I understand Chrome is not available for this platform, but Chromium and Firefox are. According to [this](https://github.com/hardkoded/puppeteer-sharp/blob/f5ebfa64e98b0de4bbc599c7fec2756fb3676c61/lib/PuppeteerSharp/IPage.cs#L890) comment PDF generation is only supported in Chrome headless.

How hard would it be to support Chromium and/or Firefox for PDF generation? I'm essentially trying to understand the challenges involved to see whether this is something I could tackle myself or not.

Thank you!

**Comment by kblok** [comment]
You should be able to generate PDFs using Chrome (not firefox) in Ubuntu. Our tests run in Ubuntu :)

**Reply by pguerra-ce** [reply]
Chrome works fine but it doesn't run in Linux Arm64, which is what I need. I understand Chromium (not Chrome) works in Linux Arm64, but PDF generation doesn't seem to work. I'm trying to understand exactly what prevents Chromium (not Chrome) from generating PDFs and what can be done about it.

Thanks.

**Reply by kblok** [reply]
>PDF generation doesn't seem to work

What are you getting?

**Reply by pguerra-ce** [reply]
Method `page.PdfDataAsync` never returns. It is stuck.

**Reply by tungdang214** [reply]
I second this. So how should we generate PDF if we have Linux ARM64 enviroment?
Anything suggestion would be helpful.
